uefi-test-runner/src/proto/rng.rs

```diff
@@ -0,0 +1,35 @@
+use uefi::prelude::*;
+use uefi::proto::rng::{Rng, RngAlgorithm};
+use uefi::table::boot::{BootServices, OpenProtocolAttributes, OpenProtocolParams};
+
+pub fn test(image: Handle, bt: &BootServices) {
+    info!("Running rng protocol test");
+
+    let handle = *bt
+        .find_handles::<Rng>()
+        .expect_success("Failed to get Rng handles")
+        .first()
+        .expect("No Rng handles");
+
+    let rng = bt
+        .open_protocol::<Rng>(
+            OpenProtocolParams {
+                handle,
+                agent: image,
+                controller: None,
+            },
+            OpenProtocolAttributes::Exclusive,
+        )
+        .expect_success("Failed to open Rng protocol");
+    let rng = unsafe { &mut *rng.interface.get() };
+
+    let mut list = [RngAlgorithm::default(); 4];
+
+    rng.get_info(&mut list).unwrap_success();
```

Comment:
You could add an assert here that the length of the list is `>=1`, since the spec requires support for at least one algorithm.

You can also log the list here, e.g. `info!("Supported rng algorithms: {:?}", ...)`. That matches the general style of the test-runner and makes it easier to see at a glance what it's doing. And same below for `get_rng`, could log `"Got random value: {:?}"`.

Reply:
Done